### PR TITLE
Upgrade leaflet-providers to 1.4.0

### PR DESCRIPTION
**Pull request by ktmud**
This upgrades leaflet-providers to 1.4.0 so we can use more map variants.

Related: #567 

```r
# make sure webshot is installed to take pictures of htmlwidgets
if (!("webshot" %in% installed.packages()[, "Package"])) {
  install.packages("webshot")
  # restart R process to fully integrate with knitr
}
reprex::reprex({
  library(leaflet)
  # insert reprex here
  leaflet() %>%
            setView(lng = -71.08920, lat = 42.34022, zoom = 12) %>%
            addProviderTiles(
                providers$CartoDB.VoyagerLabelsUnder,
                options = providerTileOptions(noWrap = TRUE))
})
```
Note that the new version of provider.js also forces HTTPS for some of the providers.

PR task list:
- [x ] Update NEWS
- [x ] Add tests (if appropriate)
- [x ] Update documentation with `devtools::document()`

No tests or additional document needed.

### CODE REVIEW

File: inst/htmlwidgets/lib/leaflet-providers/leaflet-providers.js [hunk 1]
```diff
@@ -90,7 +77,7 @@
 
 	L.TileLayer.Provider.providers = {
 		OpenStreetMap: {
-			url: '//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+			url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
```
Comment (schloerke) [inline inst/htmlwidgets/lib/leaflet-providers/leaflet-providers.js]: For url changes where only the `https` was dropped, could you please keep it removed? (including Stamen) . I've checked Onemap and they support both http and https. If they could have their protocol dropped as well, that'd be great.

There is a standing bug with RStudio 1.1 (fixed in 1.2 - daily build) that windows and linux can not load https tiles. There is some custom code to make the protocols work if no protocol is provided.  Once 1.2 is released, all custom changes to the protocols will be removed.

Comment (schloerke) [reply inst/htmlwidgets/lib/leaflet-providers/leaflet-providers.js]: Onemap http link ex: http://maps-b.onemap.sg/v3/Default/13/6458/4064.png

**Comment by schloerke** [comment]
Hi @ktmud 

Could you ...

"
 2. Ensure that you have signed the [individual](https://rstudioblog.files.wordpress.com/2017/05/rstudio_individual_contributor_agreement.pdf) or [corporate](https://rstudioblog.files.wordpress.com/2017/05/rstudio_corporate_contributor_agreement.pdf) contributor agreement as appropriate. You can send the signed copy to jj@rstudio.com.
"
Excerpt from [rstudio/httpuv package](https://github.com/rstudio/httpuv/blob/master/CONTRIBUTING.md), which you had no reason to know existed.

Please let me know when you've sent the email.  Thank you for your help!

Best,
Barret

**Comment by ktmud** [comment]
@schloerke Signed the sent. Please take a look. :)

File: inst/htmlwidgets/lib/leaflet-providers/leaflet-providers.js [hunk 2]
```diff
@@ -130,7 +118,7 @@
 					}
 				},
 				BZH: {
-					url: 'https://tile.openstreetmap.bzh/br/{z}/{x}/{y}.png',
+					url: '//tile.openstreetmap.bzh/br/{z}/{x}/{y}.png',
```
Comment (schloerke) [inline inst/htmlwidgets/lib/leaflet-providers/leaflet-providers.js]: Validated with http://tile.openstreetmap.bzh/br/7/63/43.png

**Comment by schloerke** [comment]
Note: @ktmud has signed waiver.

**Comment by ktmud** [comment]
Is this PR OK to merge?

BTW, learned about this service today: https://cla-assistant.io/
Seems like a good replacement to the email waiver form.

**Pull request merged**